### PR TITLE
Tsundere translation

### PR DESCRIPTION
**Pull request by MrMorkat**
translated the tsundere version of TillerinoBot into german
there are around 3-5 translations which i think are not perfect, i
should have made a comment next to them

EDIT: Created a new branch so not eveything im doing on my master branch is noted in here

### CODE REVIEW

File: tillerinobot/src/main/java/tillerino/tillerinobot/lang/TsundereGerman.java [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+	}
+
+	@Override
+	public String mixedNomodAndMods() {
+		return "Was soll das sein? Schrödinger's Mod? I have a recommendation, but the superposition would collapse as soon as it was observed. It's not like I like you enough to break the laws of physics anyways!"; //lol, don't know how to translate the reference so it's still understood as a reference
+	}
+
+	@Override
+	public String outOfRecommendations() {
+		return "WAS!? [https://github.com/Tillerino/Tillerinobot/wiki/FAQ#the-bot-says-its-out-of-recommendations-what-do Bist du wirklich du JEDE Empfehlung die ich dir gegeben habe durchgegangen?] Ich k-kann mir das nicht vorstellen... nun, lass sie uns erneut durchlaufen. Sieht nicht so aus als hättest du sonst was zu tun.";
```
Comment (Tillerino) [inline tillerinobot/src/main/java/tillerino/tillerinobot/lang/TsundereGerman.java]: > JEDE Empfehlung, die ich dir gegeben habe, durchgegangen

(Kommas)

File: tillerinobot/src/main/java/tillerino/tillerinobot/lang/TsundereGerman.java [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+	}
+
+	@Override
+	public String notRanked() {
+		return "Hmph. Diese Beatmap wird keinem mehr pp geben.";
```
Comment (Tillerino) [inline tillerinobot/src/main/java/tillerino/tillerinobot/lang/TsundereGerman.java]: Not perfect, but this message should no longer show up anyway.

File: tillerinobot/src/main/java/tillerino/tillerinobot/lang/TsundereGerman.java [hunk 3]
```diff
@@ -0,0 +1,392 @@
...
+
+	@Override
+	public Response optionalCommentOnNP(OsuApiUser apiUser, BeatmapMeta meta) {
+		if (Math.random() > 0.25) {
+			return new NoResponse();
+		}
+		PercentageEstimates estimates = meta.getEstimates();
+		double typicalPP = (apiUser.getPp() / 20.0);
+		if (estimates.getPPForAcc(.95) / typicalPP > 2.0) {
+			return new Message("Ist das dein Ernst!? Wenn dich diese Map nicht tötet werde ich es machen.");
```
Comment (Tillerino) [inline tillerinobot/src/main/java/tillerino/tillerinobot/lang/TsundereGerman.java]: > nicht _umbringt_, werde

Komma nicht uebersehen.

File: tillerinobot/src/main/java/tillerino/tillerinobot/lang/TsundereGerman.java [hunk 4]
```diff
@@ -0,0 +1,392 @@
...
+		//The following checks are probably redundant, but they don't hurt anyone either.
+		if (Math.random() > 0.25) {
+			return new NoResponse();
+		}
+		PercentageEstimates estimates = meta.getEstimates();
+		double typicalPP = (apiUser.getPp() / 20);
+		if (estimates.getPPForAcc(.95) / typicalPP > 2.0) {
+			return new Message("Idiot! Du wirst dich noch verletzen wenn du diese Mods benutzt!");
+		} else if (estimates.getPPForAcc(1) / typicalPP < 0.5) {
+			return new Message("Wenn du wie ein Baby behandlt werden willst, hättest du einfach fragen müssen... nein, fang einfach an zu spielen.");
```
Comment (Tillerino) [inline tillerinobot/src/main/java/tillerino/tillerinobot/lang/TsundereGerman.java]: > behand_e_lt

> Nein

(gross)

File: tillerinobot/src/main/java/tillerino/tillerinobot/lang/TsundereGerman.java [hunk 5]
```diff
@@ -0,0 +1,392 @@
...
+
+		recentRecommendations++;
+		if(recentRecommendations == 7) {
+			return new Message("Ich hab viel Freizeit. Ich würde nie Maps raussuchen weil ich dich mag... r-r-rein hypothetisch.");
+		} else if(recentRecommendations == 17) {
+			return new Message("Weißt du, es ist ein Privileg so viel mit mir zu reden, kein Recht.");
+		} else if(recentRecommendations == 37) {
+			return new Message("Wie würdest du eigentlich dieses Spiel spielen wenn ich dir nicht die ganze Zeit sagen würde wie?");
+		} else if(recentRecommendations == 73) {
+			return new Message("Ich hätte dich schon längst für Belästigung angezeigt wenn ich dich nicht lieb... Ich hab nichts gesagt.");
```
Comment (Tillerino) [inline tillerinobot/src/main/java/tillerino/tillerinobot/lang/TsundereGerman.java]: > angezeigt, wenn

(Komma)

File: tillerinobot/src/main/java/tillerino/tillerinobot/lang/TsundereGerman.java [hunk 6]
```diff
@@ -0,0 +1,392 @@
...
+	public Response optionalCommentOnRecommendation(OsuApiUser apiUser, Recommendation meta) {
+		setChanged(true);
+
+		recentRecommendations++;
+		if(recentRecommendations == 7) {
+			return new Message("Ich hab viel Freizeit. Ich würde nie Maps raussuchen weil ich dich mag... r-r-rein hypothetisch.");
+		} else if(recentRecommendations == 17) {
+			return new Message("Weißt du, es ist ein Privileg so viel mit mir zu reden, kein Recht.");
+		} else if(recentRecommendations == 37) {
+			return new Message("Wie würdest du eigentlich dieses Spiel spielen wenn ich dir nicht die ganze Zeit sagen würde wie?");
```
Comment (Tillerino) [inline tillerinobot/src/main/java/tillerino/tillerinobot/lang/TsundereGerman.java]: > spielen, wenn

(Komma)

File: tillerinobot/src/main/java/tillerino/tillerinobot/lang/TsundereGerman.java [hunk 7]
```diff
@@ -0,0 +1,392 @@
...
+		} else if(recentRecommendations == 73) {
+			return new Message("Ich hätte dich schon längst für Belästigung angezeigt wenn ich dich nicht lieb... Ich hab nichts gesagt.");
+		} else if(recentRecommendations == 173) {
+			return new Message("Kannst mich einfach nicht allein lassen, was? Ich d-denke das ist okay. Aber wag es nicht das jemandem zu erzählen!");
```
Comment (Tillerino) [inline tillerinobot/src/main/java/tillerino/tillerinobot/lang/TsundereGerman.java]: > wag es nicht, das

(Komma)

File: tillerinobot/src/main/java/tillerino/tillerinobot/lang/TsundereGerman.java [hunk 8]
```diff
@@ -0,0 +1,392 @@
...
+		this.changed = changed;
+	}
+
+	StringShuffler invalidAccuracyShuffler = new StringShuffler(rnd);
+	
+	@Override
+	public String invalidAccuracy(String acc) {
+		setChanged(true);
+
+		return invalidAccuracyShuffler.get(
```
Comment (Tillerino) [inline tillerinobot/src/main/java/tillerino/tillerinobot/lang/TsundereGerman.java]: All of these are quotes by famous people. I'd look over them again and maybe look for translations that other people have done, if you haven't already done so.

File: tillerinobot/src/main/java/tillerino/tillerinobot/lang/TsundereGerman.java [hunk 9]
```diff
@@ -0,0 +1,392 @@
...
+	}
+
+	StringShuffler optionalCommentOnLanguageShuffler = new StringShuffler(rnd);
+
+	@Override
+	public Response optionalCommentOnLanguage(OsuApiUser apiUser) {
+		setChanged(true);
+
+		return new Message(optionalCommentOnLanguageShuffler.get(
+			"[https://osu.ppy.sh/u/3135278 MrMorkat] hat mir dabei geholfen Deutsch zu lernen. Aber welche Art von Idiot will einen Tsundere Roboter auf Deutsch!? Das ist ernsthaft die dümmste Idee die ich je gehört habe!",
```
Comment (Tillerino) [inline tillerinobot/src/main/java/tillerino/tillerinobot/lang/TsundereGerman.java]: > die dümmste Idee, die

(Komma)

File: tillerinobot/src/main/java/tillerino/tillerinobot/lang/TsundereGerman.java [hunk 10]
```diff
@@ -0,0 +1,392 @@
...
+			"[https://osu.ppy.sh/u/3135278 MrMorkat] hat mir dabei geholfen Deutsch zu lernen. Das wollte ich dir immer schon in einer anderen Sprache sagen: Ich lieb... Ich hab nichts gesagt!",
+			"[https://osu.ppy.sh/u/3135278 MrMorkat] hat mir dabei geholfen Deutsch zu lernen. Aber ich hab das nur gemacht weil ich es wollte. Das hat nichts mit dir zu tun!"
```
Comment (Tillerino) [inline tillerinobot/src/main/java/tillerino/tillerinobot/lang/TsundereGerman.java]: > nur gemacht, weil

(Komma)

File: tillerinobot/src/main/java/tillerino/tillerinobot/lang/TsundereGerman.java [hunk 11]
```diff
@@ -0,0 +1,392 @@
...
+			if (invalidRecommendationParameterCount++ % 4 < 3) {
+				return unknownRecommendationParameter();
+			}
+		}
+		return "Was soll \"" + invalid + "\" bitte bedeuten!? Falls zwei Finger zu viel sind, versuch doch jeden Buchstaben zu singletappen: " + choices;
+	}
+
+	@Override
+	public String setFormat() {
+		return "Drei Worte: !set option_name value_to_set. Versuch !help falls dir drei-Wort-Sätze zu kompliziert sind."; //dunno if i should translate the example command
```
Comment (Tillerino) [inline tillerinobot/src/main/java/tillerino/tillerinobot/lang/TsundereGerman.java]: No, that's perfect. Y'know apart from the obligatory comma.

> Versuch !help, falls

File: tillerinobot/src/main/java/tillerino/tillerinobot/lang/TsundereGerman.java [hunk 12]
```diff
@@ -0,0 +1,392 @@
...
+	@Override
+	public Response welcomeUser(OsuApiUser apiUser, long inactiveTime) {
+		String username = apiUser.getUserName();
+		String greeting = "";
+		//Greetings for <4 minutes, normal, and >4 days
+		if (inactiveTime < 4 * 60 * 1000) {
+			greeting = welcomeUserShortShuffler.get(
+				"Was soll das hier sein, das Kuckuckspiel?",
+				"Das waren nicht mal fünf Minuten...",
+				"Heute hat " + username + "an seinem Verschwindungszauber geübt."
```
Comment (Tillerino) [inline tillerinobot/src/main/java/tillerino/tillerinobot/lang/TsundereGerman.java]: > Verschwindungszauber

This doesn't feel like the correct term.

Comment (MrMorkat) [inline tillerinobot/src/main/java/tillerino/tillerinobot/lang/TsundereGerman.java]: Agreed, but i couldn't think of a better one. I actually also think that Kuckuckspiel isnt quite as popular as a term for the game as peekaboo is in english

File: tillerinobot/src/main/java/tillerino/tillerinobot/lang/TsundereGerman.java [hunk 13]
```diff
@@ -0,0 +1,392 @@
...
+			);
+		} else if (inactiveTime < 4l * 24 * 60 * 60 * 1000) {
+			greeting = welcomeUserShuffler.get(
+				"Wieder da? Ich bin nur hier weil ich nix zu tun hab! Nichts weiter!",
+				"H-hey...",
+				//"♫ Home again, home again, jiggety-jig ♫ .... What?",
+				"♫ Zuhause, zuhause, lalalalala ♫ .... Was?",
```
Comment (MrMorkat) [inline tillerinobot/src/main/java/tillerino/tillerinobot/lang/TsundereGerman.java]: Ok, i'll just leave it untranslated.

File: tillerinobot/src/main/java/tillerino/tillerinobot/lang/TsundereGerman.java [hunk 14]
```diff
@@ -0,0 +1,392 @@
...
+			"Ach wirklich? Nie gehört!",
+			//"Yeah right, call me when you manage to get pp with that."
+			"Ja genau. Sag mir Bescheid wenn du damit pp erhalten hast."
+		);	
+	}
+
+	@Override
+	public String internalException(String marker) {
+		return "Huch? Warum funktioniert das nicht? Das war bestimmt dein Fehler!"
+		+ " Falls dies weiterhin vorkommt leite folgendes an [https://twitter.com/Tillerinobot @Tillerinobot] oder [http://www.reddit.com/user/tillerino /u/Tillerino]weiter: "+ marker + ".";
```
Comment (MrMorkat) [inline tillerinobot/src/main/java/tillerino/tillerinobot/lang/TsundereGerman.java]: I tried to make these more formal, as they are supposed to be taken seriously.

File: tillerinobot/src/main/java/tillerino/tillerinobot/lang/TsundereGerman.java [hunk 15]
```diff
@@ -0,0 +1,392 @@
...
+	}
+
+	StringShuffler tryWithModsShuffler = new StringShuffler(rnd);
+
+	@Override
+	public String tryWithMods() {
+		setChanged(true);
+
+		return tryWithModsShuffler.get(
+			"An idiot like you wouldn't know to try this with mods. You should thank me.",								//Need help with this one
```
Comment (MrMorkat) [inline tillerinobot/src/main/java/tillerino/tillerinobot/lang/TsundereGerman.java]: Yeah, that fits perfectly.

File: tillerinobot/src/main/java/tillerino/tillerinobot/lang/TsundereGerman.java [hunk 16]
```diff
@@ -0,0 +1,392 @@
...
+			"Ich würde sogar fast sagen, dass du hier Mods benutzen kannst, ohne wie ein kompletter Idiot auszusehen.",
+			"Vielleicht bist du hier dazu in der Lage andere Mods außer NF zu benutzen. Allerdings reden wir hier immer noch von dir."
```
Comment (MrMorkat) [inline tillerinobot/src/main/java/tillerino/tillerinobot/lang/TsundereGerman.java]: I will think about a better translation.

**Comment by MrMorkat** [comment]
Im terribly sorry for all my comma mistakes, it was 3-4 AM while i translated xD. I will fix them tomorrow as soon as i can. Thanks for reviewing and for helping with some of the translations :).

**Comment by omkelderman** [comment]
So many commits NotLikeThis Kappa

**Comment by Tillerino** [comment]
Everything good to go? (Sorry for the delay)

**Comment by MrMorkat** [comment]
everything good to go :)

**Pull request merged**